### PR TITLE
Generate pgn When ending the game

### PR DESCRIPTION
**Pull request by nocommentcode**
The PR implements the logic to generate the pgn at the end of the game.
The pgn will be generated before prompting the user for a signature so that if there are any errors, the user will be notified before signing.

### CODE REVIEW

File: packages/TorneloScoresheet/src/chessEngine/chessEngineInterface.ts [hunk 1]
```diff
@@ -67,6 +67,19 @@ export type ChessEngineInterface = {
    * @returns if the move is targeting the opposite player's piece
    */
   isOtherPlayersPiece: (fen: string, move: MoveSquares) => boolean;
+
+  /**
+   * Generates the PGN of the game
+   * @param originPgn the pgn of the even with the headers
```
Comment (gestureleft) [inline packages/TorneloScoresheet/src/chessEngine/chessEngineInterface.ts:73]: What does "even" mean in this sentence?

**Review by gestureleft — CHANGES_REQUESTED** [review]
Nice! Looks really good! I've left one small comment

File: packages/TorneloScoresheet/src/chessEngine/chessEngineInterface.ts (hunk 1 above)
Comment (nocommentcode) [reply packages/TorneloScoresheet/src/chessEngine/chessEngineInterface.ts:73]: event ** good spot

**Review by gestureleft — APPROVED** [review]
LGTM

**Pull request merged**